core/src/main/java/jenkins/model/Jenkins.java

```diff
@@ -5259,6 +5266,10 @@ private static void computeVersion(ServletContext context) {
      */
     static final String WORKSPACES_DIR_PROP = Jenkins.class.getName() + ".workspacesDir";
 
+
+    @Restricted(Beta.class)
+    public static boolean SHOW_STACK_TRACE = Boolean.getBoolean(Jenkins.class.getName() + ".SHOW_STACK_TRACE");
```

Comment:
should there be some config that automatically sets this with `jetty:run`?

Reply:
That could be great yes (separate PR anyway?)

Reply:
I think we should. We need to identify the places it should go. I expect they would all be different PRs.

Reply:
why does this need to be public?  its controlled either via a system property at startup (no need to be public for that) - or you at run time via goovy no doubt (which again does not need to be public).

Reply:
Makes testing easier. There are many other places in Jenkins that use the same pattern.

Reply:
many places causing huge issues.  do not do it untill you have played with the bytecode comparability transformer

Reply:
Changed it.

Reply:
> many places causing huge issues

@jtnord Could you elaborate for future readers and myself?

Reply:
if someone adapts a field with the same name then all classes using that field will be fudged at runtime (kind of OK).  but also any class that declares a field with the same name will be re-written.   The more classes that get rewritten the more likely you are to hit a bug in the bytecode-compatability-transformer.  and when you hit a bug in there have fun!

there is no need for these fields to be public.  runtime users can use groovy to change it (which does not care), test code can set a system property before instantiating the class, or if it wants to change it at runtime use some reflection, or make the property protected or other parts.  but adding to a public API a property to control switching removes your ability to refactor later.

Reply:
@jtnord, is your concern here only related to jenkins-test-harness? I think your concern here is more general but I'm not certain.

Reply:
my concern is you don't need public variables  don't create them.

I have not seen a need for this to be a public variable.

Reply:
It seems there is a lot of such behavior in Jenkins core. The argument is totally valid.